### ISSUE TITLE
Inaccurate docstring in `omp.pyx::determine_num_threads`

### ISSUE THREAD

**Issue by jhlegarreta** [dipy/dipy#3298]
## Description
The `omp.pyx::determine_num_threads` docstring stays that when `num_threads < 0` `the maximal number of threads minus
|num_threads + 1|` will be returned/used. It is assumed that the vertical bars mean absolute values. However, the corresponding code block reads:

https://github.com/dipy/dipy/blob/6d30ad44c0b9e2202e28e8eab5e2b15fb26260f8/dipy/utils/omp.pyx#L76-L77

Which does not match the docstring.

It does not match the behavior observed in the test either (which matches the code):

https://github.com/dipy/dipy/blob/6d30ad44c0b9e2202e28e8eab5e2b15fb26260f8/dipy/utils/tests/test_omp.py#L66-L71

The use of the vertical bars outside a math context generates warnings when generating the documentation:
https://github.com/dipy/dipy/actions/runs/9982200633/job/27587486843#step:5:893

```
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/dipy/align/streamlinear.py:docstring of dipy.align.streamlinear.StreamlineDistanceMetric.__init__:9: ERROR: Undefined substitution referenced: "num_threads + 1".
```

since it matches the Sphinx substitution syntax: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#substitutions

So the docstring should be corrected to reflect the actual behavior, or the code changed, and the substitution markups changed so that they are taken as such and reflect the correct/actual behavior.

The change was introduced in https://github.com/dipy/dipy/pull/2352.

## Way to reproduce

N/A

**Comment by jhlegarreta** [comment]
Pinging @drombas.

**Comment by jhlegarreta** [comment]
After re-reading the code, I think the docstrings are OK.

Commit de751c2 fixes the documentation issue that made me think that the documentation was inaccurate.

I will close the issue once PR #3304 gets merged

**Comment by skoudoro** [comment]
closed by #3304
